### ISSUE TITLE
Save before run

### ISSUE THREAD

**Issue by yamsellem** [TabDigital/atom-mocha-test-runner#43]
Launching a test file with `alt+ctrl+m` is not saving the current file.
So one has to `ctrl+s` and, then, `alt+ctrl+m` to launch the actual test.
Every time.

Furthermore, I'm very used to the [autosave](https://github.com/atom/autosave) package (provided with atom). 
File are saved on focus lost, so I no longer hit `ctrl+s` anymore. 
Saving the current file before launching should be a config option, don't you think?

**Comment by yamsellem** [comment]
Anyone?

**Comment by rprieto** [comment]
Sorry, completely missed that one. Agreed that auto-save before run would be handy.

So does it mean that the actual `autosave` package doesn't work, because `alt-ctrl-m` doesn't really count as lost of focus? If so, we can definitely add a `[ ] auto-save before run` option.

**Comment by yamsellem** [comment]
@rprieto you're right: `alt-crtl-m` doesn't count as a lost of focus. So an auto-save option would be great.

**Comment by yamsellem** [comment]
Is it planned for a future release?
Thanks.

**Comment by yamsellem** [comment]
Can anyone take this one?
It's been almost a year it's open.

Any idea on how do to it?
If I got clues, I may offer a pull request.

**Comment by mrstux** [comment]
👍

**Comment by mrstux** [comment]
So, if you change a test, and don't save it, and then hilite the test and run it with ctrl-alt-m, the old test runs since you didn't save it.

Would be great to auto-save on test.
